### PR TITLE
GH-37208: [R] Use currrently running R binary to compile test program (nix install)

### LINKED ISSUE #37208: [R] Arrow package install - R binary version mismatch
### Describe the bug, including details regarding any error messages, version, and platform.

Error that you might  see : `*** Unable to find libcurl and openssl`

For systems with multiple R versions installed (Posit Connect, Posit Workbench...).

Installing arrow could results in unexpected behavior.

Say you run `install.packages("arrow")` using `/opt/R/4.3.1/bin/R` but your system R version (`which R`) is `/opt/R/4.3.0/bin/R`.
Or maybe R executable is not on your path.

This script https://github.com/apache/arrow/blob/8c4941b879670ec9dc3608b77637f44fddf32178/r/tools/nixlibs.R#L200 will use the system's R instead of /opt/R/4.3.1/bin/R or R will not be found resulting in an error.

Would it be possible to locate the currently running R executable and use that instead?

```r
runner <- "`R CMD config CXX17` `R CMD config CPPFLAGS` `R CMD config CXX17FLAGS` `R CMD config CXX17STD` -E -xc++"
```

```r
r_exec <- paste(R.home(component = "bin"), "R CMD config", sep="/")
runner <- sprintf("`%s CXX17` `%s CPPFLAGS` `%s CXX17FLAGS` `%s CXX17STD` -E -xc++", r_exec, r_exec, r_exec, r_exec)
```

Thanks

### Component(s)

R

### PR DESCRIPTION
**Pull request by meztez**
### Rationale for this change

See #37208.

### What changes are included in this PR?

Use the currrently running R binary to compile test program (nix install).

### Are these changes tested?

No. Installation.

### Are there any user-facing changes?

No

* Closes: #37208

### CODE REVIEW

File: r/tools/nixlibs.R [hunk 1]
```diff
@@ -197,7 +197,9 @@ compile_test_program <- function(code) {
   # Note: if we wanted to check for openssl on macOS, we'd have to set the brew
   # path as a -I directory. But since we (currently) only run this code to
   # determine whether we can download a Linux binary, it's not relevant.
-  runner <- "`R CMD config CXX17` `R CMD config CPPFLAGS` `R CMD config CXX17FLAGS` `R CMD config CXX17STD` -E -xc++"
+  r_exec <- paste(R.home(component = "bin"), "R CMD config", sep="/")
```
Comment (kou) [inline r/tools/nixlibs.R:200]: Can we use `R_CMD_config()` here?

https://github.com/apache/arrow/blob/43c05c56b37daa93e76b94bc3e6952d56d1ea3f2/r/tools/nixlibs.R#L380-L382

**Comment by kou** [comment]
Could you use our PR template instead of removing it entirely?
(You must see it when you open this PR.)

File: r/tools/nixlibs.R (hunk 1 above)
Comment (meztez) [reply r/tools/nixlibs.R:200]: The original author did not use it.
Maybe because all 4 commands need to run at the same time?
I would not know.

Comment (kou) [reply r/tools/nixlibs.R]: @nealrichardson Can we use `R_CMD_config()` here?

**Comment by assignUser** [comment]
@github-actions crossbow submit -g r

File: r/tools/nixlibs.R (hunk 1 above)
Comment (nealrichardson) [reply r/tools/nixlibs.R]: I don't see why not, and I dug through the PR from last year when this was added and don't see any reason why it has to be this way.

File: r/tools/nixlibs.R [hunk 2]
```diff
@@ -197,7 +197,9 @@ compile_test_program <- function(code) {
   # Note: if we wanted to check for openssl on macOS, we'd have to set the brew
   # path as a -I directory. But since we (currently) only run this code to
   # determine whether we can download a Linux binary, it's not relevant.
-  runner <- "`R CMD config CXX17` `R CMD config CPPFLAGS` `R CMD config CXX17FLAGS` `R CMD config CXX17STD` -E -xc++"
+  r_exec <- file.path(R.home("bin"), "R")
+  r_cmd <- paste0("`", r_exec, " CMD config ", c("CXX17", "CPPFLAGS", "CXX17FLAGS", "CXX17STD"), "`", collapse = " ")
+  runner <- sprintf("%s -E -xc++", r_cmd)
```
Comment (nealrichardson) [inline r/tools/nixlibs.R:202]: Following @kou's suggestion: 

```suggestion
  runner <- paste(
    R_CMD_config("CXX17"),
    R_CMD_config("CPPFLAGS"),
    R_CMD_config("CXX17FLAGS"),
    R_CMD_config("CXX17STD"),
    "-E",
    "-xc++"
  )
```

**Comment by kou** [comment]
@github-actions crossbow submit -g r

**Review by kou — APPROVED** [review]
+1

**Pull request merged**